### DISCUSSION TITLE
Difficulty Converging Nonlinear Residuals with Relative Permeability (Corey/FLAC) in Draining Pit Simulation

### DISCUSSION THREAD

**Discussion by Traiwit** [idaholab/moose#28711] [Q&A Modules: Porous Flow]
Hi Porous Flow experts

I'm facing a convergence issue when using relative permeability models (Corey or FLAC) in my porous flow simulation for draining pits or excavated drives, where low saturation is expected.

![image](https://github.com/user-attachments/assets/70768e88-3cc1-49d4-b2c6-def2df57f2e3)

below is my PC block (very simple)

```
  [uobj_saturation_optionVG]
    type = PorousFlowCapillaryPressureVG
    log_extension = True
    sat_lr = 0.05
    alpha = 1e-5
    m = 0.6
  []
```

Porous flow page suggested alpha = 1e-6 but S is still way to high for 'drained' area (not realistic), so i use 1e-5

below is my Kr block (again, very simple)
```
    type = PorousFlowRelativePermeabilityFLAC
    phase = 0
    m = 2
```

and below is kr = 1 (for comparision)
```
    type = PorousFlowRelativePermeabilityConst
    phase = 0
    kr = 1.0
```

below is my `Executioner` block

```
  type = Transient   # always use 'Transient' (not 'Steady'), see [Kernels]
  solve_type = NEWTON
  # line_search = 'default'
  # line_search = 'none'
  line_search = 'bt'
  # line_search = 'basic'
  ###
  ### PETSc-options:
  petsc_options = '-snes_converged_reason ' # -snes_linesearch_monitor
  petsc_options_iname = '-pc_type -pc_hypre_type -pc_hypre_boomeramg_strong_threshold'
  petsc_options_value = ' hypre    boomeramg      0.80  '
  ###
  ### tolerances:
  nl_rel_tol = 1e-10
  nl_abs_tol = 1e3
  nl_max_its = 200
  l_tol = 1e-6
  l_max_its = 30
```
****** PROBLEM IS HERE: ******
With constant relative permeability (kr = 1), my nonlinear residual |R| can reduce from 1e13 to around 1e2 to 1e3. However, with relative permeability activated, the residuals struggle to drop below 1e6 before line search breaks down. If I turn off the line search (`line_search = 'none'`), the residual doesn't improve, it just oscillates.

does this also has anything to do with my drain surface? i'm not so sure, but i dont think so
```
    [bc_drain_tet]
      type = PorousFlowPiecewiseLinearSinkBEH
      variable = porepressure_L1
      boundary = 'moving_boundary_tet'
      pt_vals = '0 1e9'
      multipliers = '0 1e9'
      flux_function = 100
      charElLength = charElLength_M0
      PT_shift = 0.
      fluid_phase = 0
      use_mobility = true    # default is false
      use_relperm = false     # default is false
    []
```

@cpgr do you have any suggestion or do i just have to live with it? NL from 1e13 to 1e6 isnt so bad, but it would be nice to see it gets to 1e3 range

Thank you!
Traiwit

**Comment by cpgr** [comment]
I've been thinking about this and still haven't worked out what it might be...

**Reply by Traiwit** [reply]
I tried to just let `line_search = 'none'` run, it managed to converged at 96th NL (and this is one of the simpler mesh of all we got)

I guess, I will have to live with `nl_abs_tol = 1e6` + Line search, i wish it could deal with drained/dry area better

**Reply by GiudGiud** [reply]
96th NL is excessive. What does the Jacobian debugger say?
and what does the SVD monitor say about the condition number? (see troubleshooting instructions)

**Reply by cpgr** [reply]
Could it be not enough fluid can flow through the drain BC when saturation is low, which sends rel perm towards zero and makes it harder to flow? Should you set 'use_relperm = true' which I assume will scale the flux appropriately?
